### PR TITLE
Added new 'jerry_value_strict_equal' API function.

### PR DESCRIPTION
**Pull request by LaszloLango**
Added new API function to perform strict equality comparison
between 'jerry_value_t's from C code.

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/api/jerry.c [hunk 1]
```diff
@@ -782,6 +783,23 @@ jerry_value_is_undefined (const jerry_value_t value) /**< api value */
...
+ *
+ * @return true  - if the two operands are strictly equal,
+ *         false - otherwise.
+ */
+bool
+jerry_value_strict_equal (const jerry_value_t lhs, /**< first operand */
+                          const jerry_value_t rhs) /**< second operand */
+{
+  jerry_assert_api_available ();
+
```
Comment (rerobika) [inline jerry-core/api/jerry.c]: If either of the arguments is an error reference, is it able to cause any side effect? Maybe a test would be great for it.

File: tests/unit-core/test-api-strict-equal.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+int
+main (void)
+{
+  TEST_INIT ();
+
+  jerry_init (JERRY_INIT_EMPTY);
+
+  /* Test objects */
+  jerry_value_t obj_val = jerry_eval ((const jerry_char_t *) "o={x:1};o", 9, JERRY_PARSE_NO_OPTS);
```
Comment (galpeter) [inline tests/unit-core/test-api-strict-equal.c]: There should be way more tests for this api. To make it easier we could do something like this:
```c
typedef struct {
  jerry_value_t lhs;
  jerry_value_t rhs;
  bool should_be_equal; 
} test_entry;

test_entry tests[] = {
 { jerry_create_number(3.1), jerry_create_number(10), true },
 ....
};

for (int idx = 0; idx < sizeof(tests) / sizeof(test_entry); idx++) {
  bool result = jerry_value_strict_equal (tests[idx].lhs, tests[idx].rhs);
  if (result !=  tests[idx].should_be_equal) {
    fprintf (stderr, "Test case failed, idx: %d\n", idx);
    TEST_ASSERT (0);
  }
  jerry_release_value (tests[idx].lhs);
  jerry_release_value (tests[idx].rhs);
}
```
or something like this.

A few missing test cases:
* string vs any other string (which are different)
* array vs any other thing
* typedarray vs any other thing
* is this true?  `jerry_value_strict_equal(a,b)` <=> `jerry_value_strict_equal(b,a)`
* error objects
* arraybuffers
* symbols

File: docs/02.API-REFERENCE.md [hunk 3]
```diff
@@ -1630,6 +1630,41 @@ jerry_value_is_undefined (const jerry_value_t value)
 
 - [jerry_release_value](#jerry_release_value)
 
+
+## jerry_value_strict_equal
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md]: If we gave strict equal check api, then maybe we should also have the simple logic operators also (`<, >, ==`). However that would be another PR if we decide to implement it.

File: jerry-core/api/jerry.c (hunk 1 above)
Comment (galpeter) [reply jerry-core/api/jerry.c]: Also if this is a special case the documentation should mention it.

**Comment by LaszloLango** [comment]
I've updated the PR.

File: jerry-core/include/jerryscript-core.h [hunk 4]
```diff
@@ -356,6 +356,11 @@ bool jerry_value_is_promise (const jerry_value_t value);
 bool jerry_value_is_string (const jerry_value_t value);
 bool jerry_value_is_undefined (const jerry_value_t value);
 
+/*
+ * Comparison functions
+ */
+jerry_value_t jerry_strict_equal (const jerry_value_t lhs, const jerry_value_t rhs);
```
Comment (akosthekiss) [inline jerry-core/include/jerryscript-core.h]: If strict comparison is made part of the public API, shouldn't `ecma_op_abstract_equality_compare` have its API counterpart, too?

Comment (akosthekiss) [reply jerry-core/include/jerryscript-core.h]: Hm, I've missed @galpeter 's suggestion. I'm not sure I completely agree with the one function in this PR though. The < and > operators aren't necessarily needed here, but === and == are quite related, so publishing them via API should happen together, IMO.

Comment (galpeter) [reply jerry-core/include/jerryscript-core.h]: @akosthekiss To have other comparison api functions was suggested in my post above, but I had another "idea" which was not written down yet: As we introduce more and more api functions for comparisons, does it make sense to have a single function for each of them or is it possible to have a single api function which can be configured to perform different kind of comparisons?

For example something like this (do not take the naming into account just the concept):
```c
enum jerry_binary_operation {
    JERRY_OP_STRICT_EQUALS, // lhs === rhs
    JERRY_OP_EQUALS, // lhs == rhs
    JERRY_OP_NOT_EQUAL, // lhs != rhs
    JERRY_OP_LESS,  // lhs < rhs
    ...
    JERRY_OP_ADD, // lhs + rhs
};

jerry_value_t
jerry_op (jerry_binary_operation op, jerry_value_t lhs, jerry_value_t rhs);
```

I'm not sure which approach is better: having api functions for each operation or having a single one. (Maybe this discussion needs its own issue.)

Comment (zherczeg) [reply jerry-core/include/jerryscript-core.h]: I like this idea.

Comment (LaszloLango) [reply jerry-core/include/jerryscript-core.h]: @galpeter I like your idea, I will do it.
